### PR TITLE
little optimization on body parsing

### PR DESCRIPTION
**Pull request by lalop**

### CODE REVIEW

**Comment by geggleto** [comment]
:+1:

File: Slim/Http/Request.php [hunk 1]
```diff
@@ -1062,22 +1062,28 @@ public function withBody(StreamInterface $body)
...
-        if ($this->bodyParsed) {
+        if (isset($this->bodyParsed)) {
             return $this->bodyParsed;
         }
 
-        if (!$this->body) {
-            return;
-        }
-
+        $this->bodyParsed = null;
```
Comment (akrabat) [inline Slim/Http/Request.php:1199]: Surely if we've got to this line, then bodyParsed is already null?

**Comment by akrabat** [comment]
Also, can you rebase?

File: Slim/Http/Request.php (hunk 1 above)
Comment (lalop) [inline Slim/Http/Request.php:1199]: Yes but now it's set so the next time `getParsedBody` is called we will not try to parse the body.

**Comment by lalop** [comment]
@akrabat yes I can and it's done :)

File: Slim/Http/Request.php (hunk 1 above)
Comment (JoeBengalen) [inline Slim/Http/Request.php:1199]: What do you mean? Null value will still return false in the isset check (line 1195).

Comment (lalop) [inline Slim/Http/Request.php:1199]: Oh yes you are right, I didn't know that.
Damn ! Is there a way to do that whithout setting a second property to know if the body is parsed ?

File: Slim/Http/Request.php [hunk 2]
```diff
@@ -1192,24 +1192,33 @@ public function withBody(StreamInterface $body)
...
             $parsed = $this->bodyParsers[$mediaType]($body);
 
             if (!is_null($parsed) && !is_object($parsed) && !is_array($parsed)) {
                 throw new RuntimeException('Request body media type parser return value must be an array, an object, or null');
+            
+            $body = (string)$this->getBody();
+            
+            if ($body) {
+            
+                $parsed = $this->bodyParsers[$mediaType]($body);
```
Comment (akrabat) [inline Slim/Http/Request.php:1212]: I'm unclear what's going on here as this is a repeat of line 1203.

**Comment by akrabat** [comment]
Tests are failing due to a syntax error. I think it's a missing closing brace.

**Comment by akrabat** [comment]
Please reopen against the `3.x` branch when ready.

**Pull request closed without merging**